.github/workflows/linux_conditional_compilation.yml

```diff
@@ -76,44 +82,267 @@ jobs:
+        run: cmake --build ${BUILD_DIR} --parallel 8 --target benchmark_app
+
+      - name: Show ccache stats
+        run: ccache --show-stats
 
-      - name: Use OpenVINO after CC
+      - name: Run with CC-ed runtime
         run: |
           ${OPENVINO_REPO}/bin/intel64/Release/benchmark_app -niter 1 -nireq 1 \
             -m ${MODELS_PATH}/models/test_model/test_model_fp32.xml -d CPU
+
+  CPU_Functional_Tests:
+    name: CPU functional tests
+    needs: Build
+    defaults:
+      run:
+        shell: bash
+    runs-on: aks-linux-8-cores
+    container:
+      image: openvinogithubactions.azurecr.io/dockerhub/ubuntu:22.04
+    env:
+      DEBIAN_FRONTEND: noninteractive # to prevent apt-get from waiting user input
+      INSTALL_TEST_DIR: /__w/openvino/openvino/install/tests
+      PARALLEL_TEST_SCRIPT: /__w/openvino/openvino/install/tests/src/tests/test_utils/functional_test_utils/layer_tests_summary/run_parallel.py
+      PARALLEL_TEST_CACHE: /__w/openvino/openvino/install/tests/test_cache.lst
+
+    steps:
+      - name: Create Directories
+        run: mkdir -p ${INSTALL_TEST_DIR}
+
+      - name: Download OpenVINO tests package
+        uses: actions/download-artifact@v3
+        with:
+          name: openvino_tests
+          path: ${{ env.INSTALL_TEST_DIR }}
+
+      - name: Extract OpenVINO packages
+        run: |
+          pushd ${INSTALL_TEST_DIR}
+            tar -xvzf openvino_tests.tar.gz -C ${INSTALL_TEST_DIR} && rm openvino_tests.tar.gz || exit 1
+          popd
+
+      - name: Install OpenVINO dependencies
+        run: bash ${INSTALL_TEST_DIR}/scripts/install_dependencies/install_openvino_dependencies.sh -c=core -c=gpu -y
+
+      - name: Install 'actions/setup-python@v4' dependencies
+        run: apt-get install -y libssl3
+
+      - uses: actions/setup-python@v4
+        with:
+          python-version: ${{ env.PYTHON_VERSION }}
+
+      - name: Install python dependencies for run_parallel.py
+        run: python3 -m pip install -r ${INSTALL_TEST_DIR}/src/tests/test_utils/functional_test_utils/layer_tests_summary/requirements.txt
+
+      - name: Restore tests execution time
+        uses: actions/cache/restore@v3
+        with:
+          path: ${{ env.PARALLEL_TEST_CACHE }}
+          key: ${{ runner.os }}-tests-functional-cpu-stamp-${{ github.sha }}
+          restore-keys: |
+            ${{ runner.os }}-tests-functional-cpu-stamp
+
+      - name: Intel CPU plugin func tests (parallel)
+        run: python3 ${PARALLEL_TEST_SCRIPT} -e ${INSTALL_TEST_DIR}/bin/intel64/Release/ov_cpu_func_tests -c ${PARALLEL_TEST_CACHE} -w ${INSTALL_TEST_DIR} -s suite -rf 0 -- --gtest_print_time=1 --gtest_filter=*smoke*
+        timeout-minutes: 40
+
+      - name: Save tests execution time
+        uses: actions/cache/save@v3
+        if: github.ref_name == 'master'
+        with:
+          path: ${{ env.PARALLEL_TEST_CACHE }}
+          key: ${{ runner.os }}-tests-functional-cpu-stamp-${{ github.sha }}
```

Comment:
would not there be a conflict with the cache created from the `linux.yml`? I believe that they would have identical names -> one of them won't be uploaded.
If the cache from `linux.yml` is sufficient I suppose this stage could be dropped.

Reply:
agree, removed